lib/PhysicalobjectCsvImporter.class.php

```diff
@@ -421,56 +441,191 @@ public function savePhysicalobjects($data)
 
     foreach ($matches as $item)
     {
+      $timer = $this->startTimer('updateExisting');
       $this->updatePhysicalObject($item, $data);
+      $timer->add();
+    }
+
+    $saveTimer->add();
+  }
+
+  public function progressUpdate($count, $data)
+  {
+    $timer = $this->startTimer('progress');
+    $freq = $this->getOption('progressFrequency');
+
+    if (1 == $freq)
+    {
+      if (0 == $this->matchedExisting)
+      {
+        $msg = 'Row [%u/%u]: name "%s" imported (%01.2fs)';
+      }
+      else {
+        $msg = 'Row [%u/%u]: Matched and updated name "%s" (%01.2fs)';
+      }
+
+      $output = sprintf($msg,
+        $this->offset,
+        $this->rowsTotal,
+        $data['name'],
+        $this->getElapsedTime('total')
+      );
+    }
+    else if ($freq > 1 && 0 == $count % $freq)
+    {
+      $output = sprintf('Imported %u of %u rows (%01.2fs)...',
+        $count, $this->rowsTotal, $this->getElapsedTime('total'));
+    }
+
+    $timer->add();
+
+    return $output;
+  }
+
+  public function reportTimes()
+  {
+    if (!$this->getOption('debug'))
+    {
+      return sprintf('Total import time: %01.2fs',
+        $this->getElapsedTime('total')).PHP_EOL;
```

Comment:
Nitpick: I've been putting spaces around the `.` concat operator... not sure if we have a standard for that or not though.

Reply:
(PSR-2-R)[https://github.com/php-fig-rectified/fig-rectified-standards/blob/master/PSR-2-R-coding-style-guide-additions.md] recommends a space before and after the concat operator, so I agree we should use that convention.